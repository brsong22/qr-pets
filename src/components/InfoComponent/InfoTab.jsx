import React, { Component } from 'react';
import PropTypes from 'prop-types';

class InfoTab extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { tabText, infoText, onTab } = this.props;
    const show = tabText === onTab ? '' : 'hidden';
    return (
      <div className={`tabInfo ${show}`} id={`#${tabText}_tab_text`}>
        {infoText}
      </div>
    );
  }
}

InfoTab.propTypes = {
  tabText: PropTypes.string.isRequired,
  infoText: PropTypes.string,
  onTab: PropTypes.string,
};

InfoTab.defaultProps = {
  infoText: '',
  onTab: 'info',
};

export default InfoTab;
