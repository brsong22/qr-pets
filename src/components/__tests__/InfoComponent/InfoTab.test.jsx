
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import InfoTab from '../../InfoComponent/InfoTab';

describe('InfoTab', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<InfoTab tabText="test" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
