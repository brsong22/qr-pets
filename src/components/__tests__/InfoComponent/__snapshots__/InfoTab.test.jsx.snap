// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTab renders as expected 1`] = `
<div
  className="tabInfo hidden"
  id="#test_tab_text"
/>
`;
